cmd/amplifier/server/configuration.go

```diff
@@ -0,0 +1,46 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+	"log"
+)
+
+// Config is used for amplifier configuration settings
+type Configuration struct {
+	Version          string
+	Build            string
+	Port             string
+	PublicAddress    string
+	EtcdEndpoints    []string
+	ElasticsearchURL string
+	NatsURL          string
+	DockerURL        string
+	DockerVersion    string
+	EmailKey         string
+	EmailSender      string
+	SmsAccountID     string
+	SmsKey           string
+	SmsSender        string
+}
+
+// ReadConfig reads the configuration file
+func ReadConfig(config *Configuration) error {
+	// Add matching environment variables - will take precedence over config files.
+	viper.AutomaticEnv()
+
+	// Add default config file search paths in order of decreasing precedence.
+	viper.SetConfigName("amplifier")
+	viper.AddConfigPath("/etc/atomiq/")
+	if err := viper.ReadInConfig(); err != nil {
```

Comment:
One concern with this is that settings can be specified in the config file that actually shouldn't be loaded (such as version and build compile time constants) and environment variables shouldn't be used at all (nor should they override) credential settings. Credentials themselves shouldn't be in the global config but stored under home directory in a directory with permissions set to 700 and file(s) set to 600. That doesn't have to be done with this PR as long as we do get it done. I'll create an issue for it.